lib/my_obfuscate.rb

```diff
@@ -97,14 +117,34 @@ def obfuscate_bulk_insert_line(line, table_name, columns, ignore = nil)
       line
     else
       check_for_defined_columns_not_in_table(table_name, columns)
-      check_for_table_columns_not_in_definition(table_name, columns) if fail_on_unspecified_columns?
+      check_for_table_columns_not_in_definition(table_name, columns)
+
+      # Prevent errors with extra columns
+      if (unspecified_columns_behavior != :fail) && table_config
+        extra_columns = extra_column_list(table_name, columns)
+        table_config.reject! { |k,v| extra_columns.include?(k) }
```

Comment:
Good catch -- I was thinking that I wanted to mutate in-place since I do want to reject the missing columns, but was forgetting that this hash is passed in from outside the method.

I'm switching this up now.